src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -1322,51 +1316,36 @@ static inline int MPIDI_NM_mpi_rget(void *origin_addr,
 {
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_NM_MPI_RGET);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_NM_MPI_RGET);
+    MPIR_Request *sreq = NULL;
     int mpi_errno = MPI_SUCCESS;
-    size_t origin_bytes;
-    size_t offset;
 
     if (!MPIDI_OFI_ENABLE_RMA) {
         mpi_errno = MPIDI_CH4U_mpi_rget(origin_addr, origin_count, origin_datatype,
                                         target_rank, target_disp, target_count, target_datatype,
-                                        win, request);
+                                        win, &sreq);
         goto fn_exit;
     }
 
     MPIDI_CH4U_RMA_OP_CHECK_SYNC(target_rank, win);
 
-    MPIDI_Datatype_check_size(origin_datatype, origin_count, origin_bytes);
-
-    if (unlikely((origin_bytes == 0) || (target_rank == MPI_PROC_NULL))) {
-        mpi_errno = MPI_SUCCESS;
-        *request = MPIR_Request_create(MPIR_REQUEST_KIND__RMA);
-        MPIR_ERR_CHKANDSTMT((*request) == NULL, mpi_errno, MPIX_ERR_NOREQ, goto fn_fail,
-                            "**nomemreq");
-        MPIR_Request_add_ref(*request);
-        MPIDI_CH4U_request_complete(*request);
-        goto fn_exit;
-    }
-
-    if (target_rank == win->comm_ptr->rank) {
-        *request = MPIR_Request_create(MPIR_REQUEST_KIND__RMA);
-        MPIR_ERR_CHKANDSTMT((*request) == NULL, mpi_errno, MPIX_ERR_NOREQ, goto fn_fail,
-                            "**nomemreq");
-        MPIR_Request_add_ref(*request);
-        offset = win->disp_unit * target_disp;
-        mpi_errno = MPIR_Localcopy((char *) win->base + offset,
-                                   target_count,
-                                   target_datatype, origin_addr, origin_count, origin_datatype);
-        MPIDI_CH4U_request_complete(*request);
-        goto fn_exit;
-    }
-
     mpi_errno = MPIDI_OFI_do_get(origin_addr,
                                  origin_count,
                                  origin_datatype,
                                  target_rank,
-                                 target_disp, target_count, target_datatype, win, request);
+                                 target_disp, target_count, target_datatype, win, &sreq);
+
```

Comment:
Maybe it is better to follow other existing request handling code as in the same file (see `MPIDI_NM_mpi_raccumulate`). It directly passes request pointer to `do_accumulate`, and let `do_accumulate` internally create an empty request if the pointer is not NULL. For `MPIDI_NM_mpi_accumulate`, it passes a NULL request pointer, thus no request is created.

Reply:
Ok. Yes that sounds better.

Reply:
The request handling in put/get seems still different from that in accumulate. Did you address this comment ?